### PR TITLE
Sync after switching branch

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.core.runtime.jobs.Job;
+import org.sonarlint.eclipse.core.SonarLintLogger;
+import org.sonarlint.eclipse.core.internal.SonarLintCorePlugin;
+import org.sonarlint.eclipse.core.internal.engine.connected.ResolvedBinding;
+import org.sonarlint.eclipse.core.resource.ISonarLintProject;
+
+public class StorageSynchronizerJob extends Job {
```
Comment (damien-urruty-sonarsource) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java:38]: There is already a `StorageSynchronizerJob` in the `ui` module, do they relate to each other ? Could't we have only one ?

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    var subMonitor = SubMonitor.convert(monitor, projectsToSyncByConnectionId.size());
+    subMonitor.setTaskName("Sync SonarLint Storages");
+    for (Map.Entry<String, Set<String>> entry : projectsToSyncByConnectionId.entrySet()) {
+      String connectionId = entry.getKey();
+      Set<String> projectKeys = entry.getValue();
+      SonarLintCorePlugin.getServersManager().findById(connectionId).ifPresent(connection -> {
+        subMonitor.subTask("Sync SonarLint Storage for connection '" + connectionId + "'");
+        var connectionMonitor = subMonitor.newChild(1);
+        try {
+          connection.autoSync(projectKeys, connectionMonitor);
```
Comment (damien-urruty-sonarsource) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java:70]: Not directly related to this line, when we do a sync, server branches could change, shouldn't we re-resolve the branch, or evict the cache ?

**Review by damien-urruty-sonarsource — APPROVED** [review]

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java (hunk 1 above)
Comment (henryju) [reply org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java:38]: I will rename this other `PeriodicStoragesSynchronizerJob` for more clarity

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java (hunk 2 above)
Comment (henryju) [reply org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/StorageSynchronizerJob.java:70]: This is done in the `ConnectedEngineFacade` `sync` method.

**Pull request merged**